contracts/test/TestJBveBanny.t.sol

```diff
@@ -60,55 +68,131 @@ contract JBveBannyTests is TestBaseWorkflow {
+    assertEq(_jbToken.balanceOf(address(_jbveBanny), _projectId), 0);
   }
 
   function testExtendLock() public {
     mintIJBTokens();
-    uint256 _tokenId = _jbveBanny.lock(_projectOwner, 10 ether, 864000, _projectOwner, true, false);
-    (, uint256 d, uint256 lockedUntil, ,) = _jbveBanny.getSpecs(_tokenId);
-    assertEq(d, 864000);
-    evm.warp(lockedUntil * 2);
+    uint256 _tokenId = _jbveBanny.lock(_projectOwner, 10 ether, 604800, _projectOwner, true, false);
+    (, uint256 d, uint256 lockedUntil, , ) = _jbveBanny.getSpecs(_tokenId);
+    assertEq(d, 604800);
+    evm.warp(lockedUntil + 2);
+    uint256 votingPowerBeforeExtending = _jbveBanny.tokenVotingPowerAt(1, block.number);
 
     JBLockExtensionData[] memory extends = new JBLockExtensionData[](1);
-    extends[0] = JBLockExtensionData(
-      1, 8640000
-    );
+    extends[0] = JBLockExtensionData(1, 2419200);
     _tokenId = _jbveBanny.extendLock(extends)[0];
-
-    (, uint256 duration, , ,) = _jbveBanny.getSpecs(_tokenId);
-    assertEq(duration, 8640000);
+    uint256 votingPowerAfterExtending = _jbveBanny.tokenVotingPowerAt(1, block.number);
+    assert(votingPowerAfterExtending > votingPowerBeforeExtending);
+    (, uint256 duration, , , ) = _jbveBanny.getSpecs(_tokenId);
+    assertEq(duration, 2419200);
   }
 
+
+  function testRedeem() public {
+    mintIJBTokens();
+    uint256 _tokenId = _jbveBanny.lock(_projectOwner, 10 ether, 604800, _projectOwner, true, false);
+    (, , uint256 lockedUntil, , ) = _jbveBanny.getSpecs(_tokenId);
+    evm.warp(lockedUntil * 2);
+    evm.stopPrank();
+
+    evm.startPrank(address(_jbveBanny));
+    uint256[] memory _permissionIndexes = new uint256[](1);
+    _permissionIndexes[0] = JBOperations.BURN;
+    jbOperatorStore().setOperator(
+      JBOperatorData(address(_redemptionTerminal), _projectId, _permissionIndexes)
+    );
+    evm.stopPrank();
+
+    evm.startPrank(_projectOwner);
+    // adding overflow
+    _jbTerminalToken.approve(address(_redemptionTerminal), 40 ether);
+     IJBPaymentTerminal(_redemptionTerminal).addToBalanceOf(
+      _projectId,
+      40 ether,
+      address(0),
+      'Forge test',
+      new bytes(0)
+    );
+
+    JBRedeemData[] memory redeems = new JBRedeemData[](1);
+    redeems[0] = JBRedeemData(
+      _tokenId,
+      address(0),
+      1 ether,
+      payable(_projectOwner),
+      'test memo',
+      '0x69',
+      IJBRedemptionTerminal(_redemptionTerminal)
+    );
+
+    uint tokenStoreBalanceBeforeRedeem = _jbTokenStore.balanceOf(address(_jbveBanny), _projectId);
+    uint jbTerminalTokenBalanceBeforeRedeem = _jbTerminalToken.balanceOf(_projectOwner, _projectId);
+
+    _jbveBanny.redeem(redeems);
+
+    uint jbTerminalTokenBalanceAfterRedeem = _jbTerminalToken.balanceOf(_projectOwner, _projectId);
+    uint tokenStoreBalanceAfterRedeem = _jbTokenStore.balanceOf(address(_jbveBanny), _projectId);
+
+    assert(tokenStoreBalanceBeforeRedeem > tokenStoreBalanceAfterRedeem);
```

Comment:
using a redemption rate of 10000 (max rate), this should be easy to compute accurately (more than gt)
+ use DSTest assert's